test/over_react/component_declaration/flux_component_test.dart

```diff
@@ -24,33 +30,33 @@ void main() {
 
   group('FluxUiProps', () {
     test('exposes an actions getter', () {
-      var props = TestDefault();
+      var props = TestBasic();
       var testActions = new TestActions();
 
       props.actions = testActions;
 
       expect(props.actions, equals(testActions));
-      expect(props, containsPair('TestDefaultProps.actions', testActions),
+      expect(props, containsPair('TestBasicProps.actions', testActions),
           reason: 'should have an `actions` getter with the a properly-namespaced prop key');
     });
 
     test('exposes a store getter', () {
-      var props = TestDefault();
+      var props = TestBasic();
       var testStore = new TestStore();
 
       props.store = testStore;
 
       expect(props.store, equals(testStore));
-      expect(props, containsPair('TestDefaultProps.store', testStore),
+      expect(props, containsPair('TestBasicProps.store', testStore),
           reason: 'should have a `store` getter with the a properly-namespaced prop key');
     });
   });
 
-  group('FluxUiComponent', () {
+  void sharedComponentTests(BaseTestComponents testComponents) {
     test('subscribes to a single store by default', () async {
       var store = new TestStore();
-      var renderedInstance = render(TestDefault()..store = store);
-      TestDefaultComponent component = getDartComponent(renderedInstance);
+      var renderedInstance = render(testComponents.basic()..store = store);
+      dynamic component = getDartComponent(renderedInstance);
```

Comment:
These are `dynamic` since I didn't want to create a shared interface for components, since components can't implement each other due to generic type collisions